src/main.rs

```diff
@@ -304,7 +304,10 @@ async fn get_images<'a>(
 
         let mut futures = vec![];
 
-        let mut add_future_images = |image_ids: &'a str, format, scale| {
+        let mut add_future_images = |image_ids, format, scale| {
```

Comment:
Ideally, the required change would be to only remove the lifetime annotation:
```diff
-       let mut add_future_images = |image_ids: &'a str, format, scale| {
+      let mut add_future_images = |image_ids: &str, format, scale| {
```
but this unfortunately exposes another rust bug that requires such a hacky solution. See rust-lang/rust#58052 rust-lang/rust#70263

Reply:
@aliemjay thank you for this! could you add the explanation in the comments in line 308 or probably the link to this PR so it's clearer for people reading the code and easier to understand the motivations that led to this particular hack?

Reply:
created a new issue rust-lang/rust#100002

Reply:
That’s great! Thank you very much!